src/components/Map.tsx

```diff
@@ -370,36 +427,42 @@ const MapContainer = (props) => {
 		Polygon geometries, which would require wrapping all
 		the coordinates with the extend method. */
 
-    let bounds = coordinates.reduce(function (bounds, coord) {
-      return bounds.extend(coord);
-    }, new maplibregl.LngLatBounds(coordinates[0], coordinates[0]));
+    const bounds = coordinates.reduce(
+      // @ts-expect-error TODO: Figure out these types
+      (bounds, coord) => bounds.extend(coord),
+      // @ts-expect-error TODO: Figure out these types
+      new maplibregl.LngLatBounds(coordinates[0], coordinates[0])
+    );
 
     if (lines.length) {
-      map.current.fitBounds(bounds, {
+      map.current?.fitBounds(bounds, {
         padding: 100,
       });
     }
   };
 
-  const updateQueryFromParam = (start) => {
-    const url = new URL(window.location);
+  const updateQueryFromParam = (start: Coords) => {
+    const url = new URL(window.location.href);
     url.searchParams.set("from", lngLatToString(start));
     window.history.pushState({}, "", url);
 
     setHasStart(true);
     setStart(start);
   };
 
-  const updateQueryToParam = (dest) => {
-    const url = new URL(window.location);
+  const updateQueryToParam = (dest: Coords) => {
+    const url = new URL(window.location.href);
     url.searchParams.set("to", lngLatToString(dest));
     window.history.pushState({}, "", url);
 
     setHasEnd(true);
     setDest(dest);
   };
 
-  const getQuery = (start, dest) => {
+  const getQuery = (start: Coords | null, dest: Coords | null) => {
+    if (start === null || dest === null) {
```

Comment:
I do not believe that the params can be null at this point.

Reply:
Hmm, I think I tried without null and got some complaint from TypeScript. But go ahead and remove it if you're certain. Remember to do `npm run typecheck` to see what TypeScript thinks (unless you see TypeScript errors directly in your editor).